gossipsub.go

```diff
@@ -410,6 +415,14 @@ func (gs *GossipSubRouter) heartbeat() {
 			}
 		}
 
+		// if we still don't have enough peers request discovering more
+		if len(peers) < GossipSubDlo {
+			rediscoverTopic := topic
+			go func() {
+				gs.p.rediscover <- &Rediscover{rediscoverTopic, []discovery.Option{discovery.Limit(GossipSubDhi)}}
```

Comment:
it's ok to discover more peers, let's not limit to just `Dhi`.

Reply:
Given that Dhi, Dlo, etc. are not global variables instead of construction parameters I figured this was likely to be yet another global variable and I'm open to what you'd like that to be. I'd prefer there to be a limit instead of an "unlimited" discovery option since there's no explicit contract on the Discover channel even closing.

Yes, we have a context currently set to 10 seconds for Discovery, but it feels a bit awkward for waiting 10 seconds to potentially be the default (e.g. if I did a joint DHT + Rendezvous search). Having stateful clients that buffer the "extra" clients seems sufficient and since we can't process/handle more than `Dhi` peers anyway I figure let's let the discovery client worry about excess peers.

Reply:
I added a global var `GossipSubNPeers = 100` to replace `GossipSubDhi` here.

Reply:
Probably ok for now.